firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -264,6 +271,10 @@ private void initialize(Context context, User user, FirebaseFirestoreSettings se
     if (gcScheduler != null) {
       gcScheduler.start();
     }
+
+    if (indexScheduler != null && Persistence.INDEXING_SUPPORT_ENABLED) {
```

Comment:
I think we should guard the scheduler assignment based on this flag (line 265). Ideally, if indexing is off, none of the logic should even be loaded. It otherwise becomes hard to reason about the system - e.g. the constructor code of the scheduler needs to be aware that indexing might be on or off.

Reply:
Guarded based on the flag and persistence being enabled.